iocore/cluster/P_ClusterCacheInternal.h

```diff
@@ -329,14 +330,8 @@ struct CacheContinuation : public Continuation {
   inline static bool
   is_ClusterThread(EThread *et)
   {
-    int etype = ET_CLUSTER;
-    int i;
-    for (i = 0; i < eventProcessor.n_threads_for_type[etype]; ++i) {
-      if (et == eventProcessor.eventthread[etype][i]) {
-        return true;
-      }
-    }
-    return false;
+    auto &tg = eventProcessor.thread_group[ET_CLUSTER];
+    return ET_CLUSTER && std::any_of(tg._thread, tg._thread + tg._count, [et](EThread *t) { return t == et; });
```

Comment:
When is it valid that a thread in the ET_CLUSTER thread_group not be of type ET_CLUSTER?